renderdoc/driver/metal/metal_device_bridge.mm

```diff
@@ -257,17 +257,32 @@ - (MTLSizeAndAlign)heapBufferSizeAndAlignWithLength:(NSUInteger)length
 
 - (nullable id<MTLTexture>)newTextureWithDescriptor:(MTLTextureDescriptor *)descriptor
 {
-  METAL_NOT_HOOKED();
-  return [self.real newTextureWithDescriptor:descriptor];
+  return id<MTLTexture>(
+      GetWrapped(self)->newTextureWithDescriptor((MTL::TextureDescriptor *)descriptor));
 }
 
 - (nullable id<MTLTexture>)newTextureWithDescriptor:(MTLTextureDescriptor *)descriptor
                                           iosurface:(IOSurfaceRef)iosurface
                                               plane:(NSUInteger)plane
     API_AVAILABLE(macos(10.11), ios(11.0))
 {
-  METAL_NOT_HOOKED();
-  return [self.real newTextureWithDescriptor:descriptor iosurface:iosurface plane:plane];
+  NS::String *nsString = (NS::String *)[[NSThread callStackSymbols] objectAtIndex:1];
+  // Example parentCallsite string
+  //"1 QuartzCore 0x00000001b956ece8 _ZL19get_unused_drawableP20_CAMetalLayerPrivatebb + 676"
+  bool nextDrawable = false;
+  if(nsString)
+  {
+    rdcstr parentCallsite(nsString->utf8String());
+    nextDrawable = (parentCallsite.contains("CAMetalLayer") && parentCallsite.contains("drawable"));
+  }
```

Comment:
The way this is solved on other platforms is to ensure that the wrapping covers the set of things we need to care about, so we would wrap CAMetalLayer and know the entry point that way. How feasible is that? This feels very brittle, though at least the worst consequence is that the reported call is wrong.

If the whole callstack is examined, will you see the user's call to `nextDrawableTexture`? That seems at least more robust in the event that the internal implementation changes.